r18n-core/locales/cy.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module R18n
+  module Locales
+    # Welsh locale
+    class Cy < Locale
+      set(
+        title: 'Welsh',
+        sublocales: [],
+
+        week_start: :sunday,
+        wday_names: ["Dydd Sul", "Dydd Llun", "Dydd Mawrth", "Dydd Mercher", "Dydd Iau", "Dydd Gwener", "Dydd Sadwrn"],
+        wday_abbrs: %w[Sul Llu Maw Mer Iau Gwe Sad],
+
+        month_names: %w[Ionawr Chwefror Mawrth Ebrill Mai Mehefin Gorffennaf Awst Medi Hydref Tachwedd Rhagfyr],
+        month_abbrs: %w[Ion Chw Maw Ebr Mai Meh Gor Aws Med Hyd Tac Rha],
+
+        date_format: '%Y-%m-%d',
+        full_format: '%e o %B',
+        year_format: '_, %Y',
+
+        number_decimal: '.',
+        number_group:   ','
+      )
+
+      def ordinalize(n)
+          case n % 10
+            when 1 then "#{n}af"
+            when 2 then "#{n}il"
+            when 3 then "#{n}ydd"
+            when 4 then "#{n}ydd"
+            when 11 then "#{n}eg"
+            when 13 then "#{n}eg"
+            when 14 then "#{n}eg"
+            when 16 then "#{n}eg"
+            when 17 then "#{n}eg"
+            when 19 then "#{n}eg"
+            when 21..31 then "#{n}ain"
```

Comment:
It's nice, but I think it can be done also for `eg` and `ydd` (every repeated value).

Reply:
I think this code will be simpler than tricky math

Reply:
> I think this code will be simpler than tricky math

Do you think that Ranges are tricky math? Maybe cycles are tricky math too? I don't understand this term, I just want to reduce the code.

Reply:
Oh, I see: it's not in order. I'm sorry.